src/main/java/no/nav/fpsak/tidsserie/StandardCombinators.java

```diff
@@ -137,6 +145,48 @@ public static LocalDateSegment<String> concat(LocalDateInterval dateInterval,
         return new LocalDateSegment<>(dateInterval, (lv == null ? "" : lv) + (rv == null ? "" : rv));
     }
 
+    /**
+     * Basic combinator som tar minste verdi, evt lhs dersom like
+     */
+    public static <V extends Comparable<? super V>> LocalDateSegment<V> min(LocalDateInterval dateInterval,
+                                                                            LocalDateSegment<V> lhs,
+                                                                            LocalDateSegment<V> rhs) {
+        if (lhs != null && rhs != null) {
+            var least = lhs.getValue().compareTo(rhs.getValue()) <= 0 ? lhs.getValue() : rhs.getValue();
+            return new LocalDateSegment<>(dateInterval, least);
+        }
+        return lhs == null ? new LocalDateSegment<>(dateInterval, rhs.getValue()) : new LocalDateSegment<>(dateInterval, lhs.getValue());
+    }
+
+    /**
+     * Basic combinator som tar største verdi, evt lhs dersom like
+     */
+    public static <V extends Comparable<? super V>> LocalDateSegment<V> max(LocalDateInterval dateInterval,
+                                                                            LocalDateSegment<V> lhs,
+                                                                            LocalDateSegment<V> rhs) {
+        if (lhs != null && rhs != null) {
+            var greatest = lhs.getValue().compareTo(rhs.getValue()) >= 0 ? lhs.getValue() : rhs.getValue();
+            return new LocalDateSegment<>(dateInterval, greatest);
+        }
+        return lhs == null ? new LocalDateSegment<>(dateInterval, rhs.getValue()) : new LocalDateSegment<>(dateInterval, lhs.getValue());
+    }
+
+    /**
+     * Basic combinator som slår sammen Liste-verdier til en liste
+     */
+    public static <V> LocalDateSegment<List<V>> concatLists(LocalDateInterval dateInterval,
```

Comment:
Denne kan gi duplikate verdier i liste, er det greit?
Kan  kanskje trenge noe som gir unike verdier sortert også for kompletthet

Reply:
Nå vet vi ikke om input har unike lister heller. Jeg funderte litt på å ha en concatToSet og å ta inn Collection<V> og converte til list/set
Deretter lurte jeg på hvor vanlig det er med  custom equals/hash-metoder
Jeg endte med å tenke at List ikke har duplikathåndtering og at den kan gjøres av den som bruker concatLists ...

Reply:
Kanskje ha en conflate?

Reply:
Som i  <T,V> LocalDateSegment<List> conflateLists(LocalDateInterval, LocalDateSegment<T>, LocalDateSegment<V>) ?
Der har vi bothValues som man senere kan sende gjennom filterValue og mapValue

Reply:
Tanken var mest en utility der T=V

Reply:
I k9-sak har vi definert 2 kombinatorer som kanskje også bør løftes inn her:

public static <T> LocalDateSegment<Set<T>> union(LocalDateInterval dateInterval, LocalDateSegment<Set<T>> lhs, LocalDateSegment<Set<T>> rhs) {
        Set<T> lv = lhs == null ? null : lhs.getValue();
        Set<T> rv = rhs == null ? null : rhs.getValue();

        Set<T> union;
        if (lv == null) {
            union = rv;
        } else if (rv == null) {
            union = lv;
        } else {
            union = new HashSet<>();
            union.addAll(lv);
            union.addAll(rv);
        }
        return new LocalDateSegment<>(dateInterval, union);
    }

    public static <T> LocalDateSegment<Set<T>> minus(LocalDateInterval dateInterval, LocalDateSegment<Set<T>> lhs, LocalDateSegment<Set<T>> rhs) {
        Set<T> lv = lhs == null ? null : lhs.getValue();
        Set<T> rv = rhs == null ? null : rhs.getValue();

        Set<T> resultat;
        if (lv == null) {
            resultat = null;
        } else if (rv == null) {
            resultat = lv;
        } else {
            resultat = new HashSet<>(lv);
            for (T t : rv) {
                //HashSet støtter null, men ikke i removeAll
                resultat.remove(t);
            }
        }
        return new LocalDateSegment<>(dateInterval, resultat);
    }

Reply:
Jeg satt og lurte på concatSets. Spm her om du ønsker union, minus, intersection @tendestad ?

Reply:
Det er ganske fort gjort å legge til.

Reply:
Utvidet med union, intersection, difference uten å finlese ZF